### DISCUSSION TITLE
Circumvent 99 Track Protection by requesting correct chapter

### DISCUSSION THREAD

**Discussion by koeberlue** [automatic-ripping-machine/automatic-ripping-machine#967] [Ideas]
Right now it is not possible to rip DVDs with 99 Track Protection. After searching for a solution a little bit, there seems to be a manual solution for this. I would like to discuss an idea to integrate this solution into ARM.

## Manual solution

According to [this superuser post](https://superuser.com/a/81878) you can open the movie with VLC (or any other DVD playing software), start the movie from the menu and then write down the chapter number that is being played. You can then use this chapter number to RIP the DVD with Handbrake.

# Suggested semi-automatic solution

If ARM detects a movie with 99 Track Protection, it could allow the user to manually enter the chapter that the user wants to rip. 

So as a user I would have a workflow like this:

1. Insert DVD
1. Get notified of Track 99 Protection by ARM
1. Eject DVD manually
1. Insert DVD on different device
1. Open with VLC
1. Note down correct chapter number
1. Insert DVD on PC running ARM again
1. Enter correct chapter number in Web UI
    1. ARM now rips the movie successfully

Although as a user I would have to some manual steps, I can still use all the features of ARM and reuse the configuration that I already entered.

# Suggested automatic solution

If ARM detects a movie with 99 Track Protection, it could automatically query the ARM API for the correct chapter. If it is not found, fall back to the semi-automatic solution above.

So as a user I would have a workflow like this:

1. Insert DVD
    1. ARM detects a Track 99 Protection
    1. ARM queries the ARM API for the correct chapter to rip
    2. ARM now rips the movie successfully
  
Of course we would have to add this feature to the ARM API and fill it with the correct data.  

---

I would love to see a change like this because I have a couple of movies that I did not rip because of 99 Track Protection. Of course I could go the manual way, but then I would have to replicate all configurations, copy the resulting movie to my NAS manually, fix all permissions etc.

Any thoughts on this? Does this sound feasible?

**Comment by cmbernard333** [comment]
Having recently run into this myself, would love to see this done automatically.

EDIT: what is the ARM API being referred to here?
